test/e2e/multi-stage/e2e_test.go

```diff
@@ -0,0 +1,118 @@
+			args:    []string{"--target=success", framework.RemotePullSecretFlag(t)},
+			env:     []string{defaultJobSpec},
+			success: true,
+			output:  []string{"Container test in pod success completed successfully"},
+		},
+		{
+			name:    "without references",
+			args:    []string{"--unresolved-config=config.yaml", "--target=without-references"},
+			env:     []string{defaultJobSpec},
+			success: true,
+			output:  []string{"Container test in pod produce-content completed successfully", "Container test in pod consume-and-validate completed successfully", "Container test in pod step-with-imagestreamtag completed successfully"},
+		},
+		{
+			name:    "with references",
+			args:    []string{"--unresolved-config=config.yaml", "--target=with-references"},
+			env:     []string{defaultJobSpec},
+			success: true,
+			output:  []string{"Container test in pod step completed successfully"},
+		},
+		{
+			name:    "with references via env",
+			args:    []string{"--target=with-references"},
+			env:     []string{defaultJobSpec, "UNRESOLVED_CONFIG=config.yaml"},
+			success: true,
+			output:  []string{"Container test in pod step completed successfully"},
+		},
+		{
+			name:    "skipping on success",
+			args:    []string{"--unresolved-config=config.yaml", "--target=skip-on-success"},
+			env:     []string{defaultJobSpec},
+			success: true,
+			output:  []string{`Skipping optional step "skip-on-success-skip-on-success-post-step"`},
+		},
+		{
+			name:    "step with timeout",
+			args:    []string{"--unresolved-config=config.yaml", "--target=timeout"},
+			env:     []string{defaultJobSpec},
+			success: true,
+			output: []string{`"timeout" pod "time	out-timeout" exceeded the configured timeout activeDeadlineSeconds=120`},
+		},
+		{
+			name:    "step with dependencies",
+			args:    []string{"--unresolved-config=dependencies.yaml", "--target=with-dependencies"},
+			env:     []string{depsJobSpec},
+			success: true,
+			output:  []string{`Pod with-dependencies-depend-on-stuff succeeded`},
+		},
+		{
+			name:    "step with cli",
+			args:    []string{"--unresolved-config=dependencies.yaml", "--target=with-cli"},
+			env:     []string{depsJobSpec},
+			success: true,
+			output:  []string{`Container inject-cli in pod with-cli-use-cli completed successfully`},
+		},
+		{
+			name:    "step with best effort post-step",
+			args:    []string{"--unresolved-config=best-effort.yaml", "--target=best-effort-success"},
+			env:     []string{defaultJobSpec},
+			success: true,
+			output:  []string{`Pod best-effort-success-failure is running in best-effort mode`},
+		},
+		{
+			name:    "step without best effort post-step failure",
+			args:    []string{"--unresolved-config=best-effort.yaml", "--target=best-effort-failure"},
+			env:     []string{defaultJobSpec},
+			success: false,
+			output:  []string{`could not run steps: step best-effort-failure failed`},
+		},
+	}
+
+	for _, tc := range testCases {
+		testCase := tc
```

Comment:
nit: Copying this into a different variable is extremely prone to make ppl in the future accidentally use the wrong one, it would be better to shadow it instead to not make that possible

Reply:
There was *something* goofy with not copying it ... I can't remember